### DISCUSSION TITLE
Is foreign key being used by bun?

### DISCUSSION THREAD

**Discussion by davendu** [uptrace/bun#264] [Q&A]
For example, if I have a model like [this](https://github.com/uptrace/bun/blob/5d7e5c995d3647e172cb348754c7b89dab63beab/internal/dbtest/orm_test.go#L387):
```go
	type Item struct {
		ID        int64
		CreatedAt time.Time `bun:",notnull,nullzero"`
		UpdatedAt time.Time `bun:",notnull,nullzero"`
	}

	type Order struct {
		ID    int64
		Items []Item `bun:"m2m:order_to_items"`
	}

	type OrderToItem struct {
		OrderID   int64     `bun:",pk"`
		Order     *Order    `bun:"rel:has-one,join:order_id=id"`
		ItemID    int64     `bun:",pk"`
		Item      *Item     `bun:"rel:has-one,join:item_id=id"`
		CreatedAt time.Time `bun:",notnull,nullzero"`
	}
```

And I only use `db.NewCreateTable()`, without using `ForeignKey()`. Then will Bun use foreign key automatically since there are `rel:has-one`?

**Comment by vmihailenco** [comment]
Bun does not automatically detect foreign keys. We have https://github.com/uptrace/bun/issues/197 to track this, but no work has been done in that direction.

**Comment by majidmvulle** [comment]
> Bun does not automatically detect foreign keys. We have #197 to track this, but no work has been done in that direction.

@vmihailenco I am in a sort of similar predicament. I have found this method https://github.com/uptrace/bun/blob/master/query_table_create.go#L88 that is part of the `CreateTableQuery`, is there an example of how to set this up in the go migrations?

**Reply by vmihailenco** [reply]
See https://bun.uptrace.dev/guide/query-create-table.html (the example with hooks)

**Reply by majidmvulle** [reply]
> See https://bun.uptrace.dev/guide/query-create-table.html (the example with hooks)

Thank you @vmihailenco , I should read the documentation better
